### PR TITLE
decouple arch from the build

### PR DESCRIPTION
**Pull request by vlrpl**
this is a preparation to complete  #373 
Only an arm `vmlinux.h` will be left for implementing the non-optional bullets.

### CODE REVIEW

**Review by atenart — DISMISSED** [review]
One suggestion: using `ARCH/vmlinux.h` instead of `vmlinux-ARCH.h`. But not a blocker, so I'm approving this and you can either merge it or make the change :)

**Comment by vlrpl** [comment]
> One suggestion: using `ARCH/vmlinux.h` instead of `vmlinux-ARCH.h`. But not a blocker, so I'm approving this and you can either merge it or make the change :)

I decided to go with `-arch` suffix because I don't expect other arch-specific files, so there was no strong opinion. 
Anyways, given there's a preference, I renamed the file and put it under `arc/` dir.

File: .gitattributes [hunk 1]
```diff
@@ -1,2 +1,2 @@
-vmlinux.h linguist-vendored
+vmlinux*.h linguist-vendored
```
Comment (atenart) [inline .gitattributes:1]: We probably need to update this as well.

Comment (vlrpl) [reply .gitattributes:1]: right. Should work :) but it is unneeded now. Thanks

**Review by atenart — APPROVED** [review]
LGTM, thanks!

**Pull request merged**